cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableDataLogRepairCommand.java

```diff
@@ -0,0 +1,914 @@
+     */
+    @VisibleForTesting
+    List<LogEditOperation> getDurableLogEditsFromUser() {
+        List<LogEditOperation> durableLogEdits = new ArrayList<>();
+        boolean finishInputCommands = false;
+        while (!finishInputCommands) {
+            try {
+                final String operationTpe = getStringUserInput("Select edit action on DurableLog: [delete|add|replace]");
+                switch (operationTpe) {
+                    case "delete":
+                        long initialOpId = getLongUserInput("Initial operation id to delete? (inclusive)");
+                        long finalOpId = getLongUserInput("Initial operation id to delete? (exclusive)");
+                        durableLogEdits.add(new LogEditOperation(LogEditType.DELETE_OPERATION, initialOpId, finalOpId, null));
+                        break;
+                    case "add":
+                        initialOpId = getLongUserInput("At which Operation sequence number would you like to add new Operations?");
+                        do {
+                            durableLogEdits.add(new LogEditOperation(LogEditType.ADD_OPERATION, initialOpId, initialOpId, createUserDefinedOperation()));
+                            output("You can add more Operations at this sequence number or not.");
+                        } while (confirmContinue());
+                        break;
+                    case "replace":
+                        initialOpId = getLongUserInput("What Operation sequence number would you like to replace?");
+                        durableLogEdits.add(new LogEditOperation(LogEditType.REPLACE_OPERATION, initialOpId, initialOpId, createUserDefinedOperation()));
+                        break;
+                    default:
+                        output("Invalid operation, please select one of [delete|add|replace]");
+                }
+                checkDurableLogEdits(durableLogEdits);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                ex.printStackTrace();
+            } catch (AssertionError ex) {
+                // Last input was incorrect, so remove it.
+                durableLogEdits.remove(durableLogEdits.size() - 1);
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                ex.printStackTrace();
+            }
+            output("You can continue adding edits to the original DurableLog.");
+            finishInputCommands = !confirmContinue();
+        }
+
+        return durableLogEdits;
+    }
+
+    /**
+     * Guides the user to generate a new {@link Operation} that will eventually modify the Original Log.
+     *
+     * @return New {@link Operation} to be added in the Original Log.
+     */
+    @VisibleForTesting
+    Operation createUserDefinedOperation() {
+        Operation result;
+        final String operations = "[DeleteSegmentOperation|MergeSegmentOperation|MetadataCheckpointOperation|" +
```

Comment:
We'd want to thoroughly document these options and the entire syntax/workflow for each of them. I can see an admin that isn't comfortable with the code would be reluctant to making any changes here.

Reply:
I think that this command (and perhaps the other DR commands) require a separate documentation page describing in depth their operation, workflow and so on. Will create it and complete the part related to this command.

Reply:
Committed documentation for that.